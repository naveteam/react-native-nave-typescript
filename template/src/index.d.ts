import '@react-navigation/native';

// Override the theme in react native navigation to accept our custom theme props.
declare module '@react-navigation/native' {
  export type ExtendedTheme = {
    dark: boolean;
    colors: {
      primary: string;
      secondary: string;
      white: string;
      black: string;
      error: string;
      warning: string;
      success: string;
      background: string;
      card: string;
      text: string;
      border: string;
      notification: string;
      gray: {
        n50: string;
        n100: string;
        n200: string;
        n300: string;
        n400: string;
        n500: string;
        n600: string;
        n700: string;
        n800: string;
        n900: string;
      };
    };
  };
  export function useTheme(): ExtendedTheme;
}

import 'styled-components';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: {
      white: string;
      black: string;
      error: string;
      warning: string;
      success: string;
      primary: string;
      secondary: string;
      gray: {
        n50: string;
        n100: string;
        n200: string;
        n300: string;
        n400: string;
        n500: string;
        n600: string;
        n700: string;
        n800: string;
        n900: string;
      };
    };
  }
}
