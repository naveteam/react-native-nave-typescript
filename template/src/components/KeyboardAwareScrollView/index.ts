export { default as KeyboardAwareScrollView } from './KeyboardAwareScrollView';
