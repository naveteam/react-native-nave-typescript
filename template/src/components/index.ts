export * from './Column';
export * from './Button';
export * from './Row';
export * from './Text';
export * from './Input';
export * from './KeyboardAwareScrollView';
