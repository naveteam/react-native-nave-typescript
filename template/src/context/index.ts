export { default as AppProviders } from './providers';
export * from './user';
