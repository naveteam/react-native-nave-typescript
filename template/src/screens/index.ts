export * from './Home';
export * from './Form';
