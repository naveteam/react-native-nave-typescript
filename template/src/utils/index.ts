export * from './auth';
export * from './yup-schemas';
