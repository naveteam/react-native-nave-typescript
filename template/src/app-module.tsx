import React, { FC } from 'react';
import { StatusBar } from 'react-native';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { NavigationContainer, DefaultTheme, ExtendedTheme } from '@react-navigation/native';
import { ThemeProvider } from 'styled-components/native';

import { AppProviders } from 'src/context';

import { theme } from 'src/theme';

import Routes from '../Routes';

if (__DEV__) {
  import('../ReactotronConfig').then(() => console.log('Reactotron Configured'));
}

const App: FC = () => {
  const MyTheme: ExtendedTheme = {
    ...DefaultTheme,
    colors: {
      ...DefaultTheme.colors,
      ...theme.colors
    }
  };

  return (
    <SafeAreaProvider>
      <NavigationContainer theme={MyTheme}>
        <Routes />
      </NavigationContainer>
    </SafeAreaProvider>
  );
};

export default (): JSX.Element => (
  <ThemeProvider theme={theme}>
    <AppProviders>
      <StatusBar backgroundColor={theme.colors.secondary} barStyle='dark-content' />

      <App />
    </AppProviders>
  </ThemeProvider>
);
