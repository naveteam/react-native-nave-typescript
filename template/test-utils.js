import React, { ReactNode } from 'react';
import { render } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import { theme } from 'src/theme';

const Providers = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui?: any, options?: any) => render(ui, { wrapper: Providers, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
