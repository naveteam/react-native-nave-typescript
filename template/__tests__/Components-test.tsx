import React from 'react';
import { render, fireEvent } from '../test-utils';

import { Button } from 'src/components';

describe('Button', () => {
  const buttonText = 'Test button';
  const buttonTestID = 'BUTTON_TEST_ID';

  const onPress = jest.fn();

  const MockButton = <Button testID={buttonTestID} text={buttonText} onPress={onPress} />;

  it('Should render a button, and matches with snapshot', () => {
    const { toJSON } = render(MockButton);

    expect(toJSON()).toMatchSnapshot();
  });

  it('Should call the onPress action from rendered button', () => {
    const { getByTestId } = render(MockButton);

    const button = getByTestId(buttonTestID);

    fireEvent.press(button);

    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
