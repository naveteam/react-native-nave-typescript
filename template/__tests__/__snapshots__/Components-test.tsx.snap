// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render a button, and matches with snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#6600CA",
      "borderColor": "#6600CA",
      "borderRadius": 4,
      "borderWidth": 1,
      "justifyContent": "center",
      "marginBottom": 5,
      "marginTop": 5,
      "minHeight": 39,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 8,
      "paddingRight": 8,
      "paddingTop": 8,
      "width": "100%",
    }
  }
  testID="BUTTON_TEST_ID"
>
  <View
    alignItems="center"
    flexDirection="row"
    justifyContent="center"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      fontWeight={700}
      style={
        Array [
          Object {
            "fontSize": 14,
            "fontWeight": "700",
            "lineHeight": 17,
          },
        ]
      }
      variant="small"
    >
      Test button
    </Text>
  </View>
</View>
`;
